cmd/agent/initiate/initiate_windows.go

```diff
@@ -1,64 +1,52 @@
 //go:build windows
 // +build windows
 
-package main
+package initiate
 
+//We must keep this package clear of importing any large packages so that we can initialise the windows service ASAP
 import (
-	"flag"
 	"log"
-	"os"
-
-	util_log "github.com/cortexproject/cortex/pkg/util/log"
-	"github.com/go-kit/log/level"
-	"github.com/grafana/agent/pkg/config"
-	"github.com/grafana/agent/pkg/server"
 
 	"golang.org/x/sys/windows/svc"
 )
 
+// Default name for the Grafana Agent under Windows
+const (
+	ServiceName = "Grafana Agent"
+)
+
+func init() {
+	// If Windows is trying to run as a service, go through that
+	// path instead.
+	if IsWindowsService() {
+		go func() {
+			err := svc.Run(ServiceName, &AgentService{stopCh: ServiceExit})
+			if err != nil {
+				log.Fatalf("Failed to start service: %v", err)
+			}
+		}()
+	}
+}
```

Comment:
IMO this isn't an ideal situation to be in; moving logic to `init` functions is error prone and generally not fun to maintain. 

I recognize that this was the solution used in prometheus-community/windows_exporter#1047, and for a short-term solution it might be okay. 

Long-term, though, it's best for windows_exporter (and any of its transitive dependencies causing this issue) to move away from init functions and towards lazy loading so we can remove the need for this. This is something we can assist with since we're a direct downstream dependency.

Reply:
I don't disagree, This is definitely a tactical fix thought up by someone with limited Go skills, but if nothing else im hoping this effort has helped articulate _what_ the issue is on windows machines for more competent contributors to come up with a more suitable strategic fix :)

Reply:
I will give it another run-through today/tomorrow and see if I find any oddness with using the init. It does solve a problem a lot of our users have and the PR is very much appreciated @jammiemil.